admin/model_api_error.go

```diff
@@ -219,3 +219,14 @@ func (o *ApiError) HasReason() bool {
 func (o *ApiError) SetReason(v string) {
 	o.Reason = &v
 }
+
+func (a *ApiError) GetContextualError(ctx context.Context) string {
+    baseError := a.Error()
+
+    // Check if context has request ID
+    if requestID, ok := ctx.Value("X-Request-ID").(string); ok && requestID != "" {
+        return fmt.Sprintf("%s (RequestID: %s)", baseError, requestID)
+    }
+
+    return baseError
+}
```

Comment:
Is there a reason we want to place this utility logic within ApiError model? Potentially this can be simplified to a utility function separate from ApiError which simply obtains the RequestID from the Go context.